stan/math/prim/functor/apply_scalar_binary.hpp

```diff
@@ -182,14 +182,15 @@ inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
  * @param y Scalar input to which operation is applied.
  * @param f functor to apply to Eigen and scalar inputs.
  * @return Eigen object with result of applying functor to inputs.
- *
- * Note: The return expresssion needs to be evaluated, otherwise the captured
- *         function and scalar fall out of scope.
  */
 template <typename T1, typename T2, typename F, require_eigen_t<T1>* = nullptr,
           require_stan_scalar_t<T2>* = nullptr>
-inline auto apply_scalar_binary(const T1& x, const T2& y, const F& f) {
-  return x.unaryExpr([&f, y](const auto& v) { return f(v, y); });
+inline auto apply_scalar_binary(T1&& x, T2&& y, F&& f) {
+  return make_holder(
+      [](auto& f2, auto& x2, auto& y2) {
+        return x2.unaryExpr([&f2, y2](const auto& v) { return f2(v, y2); });
+      },
```

Comment:
I'd call these `_inner` instead of just adding a 2 to the end. Just shows it's the same thing called from an inner function
```suggestion
      [](auto& f_inner, auto& x_inner, auto& y_inner) {
        return x_inner.unaryExpr([&f_inner, y_inner](const auto& val) { return f2(val, y_inner); });
      },
```

Reply:
Also is there a reason this had to change?

Reply:
When calling `apply_scalar_ternary` with at least one scalar, the scalar is wrapped within the functor and then the remaining two inputs and the functor with scalar captured is passed to `apply_scalar_binary`. The downside was that the captured scalar was then falling out of scope when the functor was passed by reference to `unaryExpr` here. 

Instead of the `make_holder` here, I've changed this to just pass the functor by value, which seems to fix the issue